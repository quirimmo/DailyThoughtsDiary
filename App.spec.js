import 'react-native';
import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

const mockCreateTables = jest.fn();
jest.mock('./src/services/SQLiteProxy', () => ({
	getInstance: () => ({ createTables: mockCreateTables })
}));
jest.mock('moment', () => () => ({
	format: () => '2018–01–30T12:34:56+00:00'
}));

import App from './App';

describe('App snapshot', () => {
	beforeEach(() => {
		jest.useFakeTimers();
		NavigationTestUtils.resetInternalState();
		jest.clearAllMocks();
		jest.clearAllTimers();
	});

	it('renders the loading screen', async () => {
		const tree = renderer.create(<App />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders the root without loading screen', async () => {
		const tree = renderer.create(<App skipLoadingScreen />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	describe('componentDidMount', () => {
		let appInstance;
		beforeEach(() => {
			appInstance = renderer.create(<App />).getInstance();
		});

		it('should call che createTables method of SQLiteProxy', () => {
			appInstance.componentDidMount();
			expect(mockCreateTables).toHaveBeenCalled();
		});
	});
});
