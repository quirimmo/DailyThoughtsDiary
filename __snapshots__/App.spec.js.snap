// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders the loading screen 1`] = `null`;

exports[`App snapshot renders the root without loading screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "opacity": 1,
              },
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                    "overflow": "hidden",
                  },
                  Object {
                    "backgroundColor": "#000",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  accessibilityElementsHidden={false}
                  collapsable={undefined}
                  pointerEvents="auto"
                  style={
                    Object {
                      "backgroundColor": "#E9E9EF",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "paddingTop": 64,
                      "position": "absolute",
                      "right": 0,
                      "shadowColor": "black",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0.2,
                      "shadowRadius": 5,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#fff",
                        "flex": 1,
                        "padding": 10,
                      }
                    }
                  >
                    <View>
                      <View
                        style={
                          Object {
                            "marginBottom": 10,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={undefined}
                        >
                          Symptoms:
                        </Text>
                        <TextInput
                          allowFontScaling={true}
                          onChangeText={[Function]}
                          placeholder="Type symptoms..."
                          style={
                            Object {
                              "fontSize": 16,
                              "height": 50,
                            }
                          }
                          value={undefined}
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "marginBottom": 10,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={undefined}
                        >
                          Location:
                        </Text>
                        <TextInput
                          allowFontScaling={true}
                          onChangeText={[Function]}
                          placeholder="Type where you are..."
                          style={
                            Object {
                              "fontSize": 16,
                              "height": 50,
                            }
                          }
                          value={undefined}
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "marginBottom": 10,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={undefined}
                        >
                          Thoughts:
                        </Text>
                        <TextInput
                          allowFontScaling={true}
                          onChangeText={[Function]}
                          placeholder="Type thoughts..."
                          style={
                            Object {
                              "fontSize": 16,
                              "height": 50,
                            }
                          }
                          value={undefined}
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "marginBottom": 10,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "fontSize": 18,
                            }
                          }
                        >
                          Time:
                        </Text>
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "flex": 0.5,
                                "flexDirection": "row",
                              }
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                Object {
                                  "fontSize": 16,
                                }
                              }
                            >
                              21:06
                            </Text>
                          </View>
                          <View
                            style={
                              Object {
                                "flex": 0.5,
                              }
                            }
                          >
                            <View
                              accessibilityComponentType="button"
                              accessibilityLabel={undefined}
                              accessibilityTraits={
                                Array [
                                  "button",
                                ]
                              }
                              accessible={true}
                              collapsable={undefined}
                              hasTVPreferredFocus={undefined}
                              hitSlop={undefined}
                              isTVSelectable={true}
                              nativeID={undefined}
                              onLayout={undefined}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                              testID={undefined}
                              tvParallaxProperties={undefined}
                            >
                              <View
                                style={
                                  Array [
                                    Object {},
                                  ]
                                }
                              >
                                <Text
                                  accessible={true}
                                  allowFontScaling={true}
                                  disabled={undefined}
                                  ellipsizeMode="tail"
                                  style={
                                    Array [
                                      Object {
                                        "color": "#007AFF",
                                        "fontSize": 18,
                                        "padding": 8,
                                        "textAlign": "center",
                                      },
                                    ]
                                  }
                                >
                                  Change Time
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "marginTop": 30,
                          }
                        }
                      >
                        <View
                          accessibilityComponentType="button"
                          accessibilityLabel={undefined}
                          accessibilityTraits={
                            Array [
                              "button",
                              "disabled",
                            ]
                          }
                          accessible={true}
                          collapsable={undefined}
                          hasTVPreferredFocus={undefined}
                          hitSlop={undefined}
                          isTVSelectable={true}
                          nativeID={undefined}
                          onLayout={undefined}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                          testID={undefined}
                          tvParallaxProperties={undefined}
                        >
                          <View
                            style={
                              Array [
                                Object {},
                                Object {},
                              ]
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              disabled={true}
                              ellipsizeMode="tail"
                              style={
                                Array [
                                  Object {
                                    "color": "#007AFF",
                                    "fontSize": 18,
                                    "padding": 8,
                                    "textAlign": "center",
                                  },
                                  Object {
                                    "color": "#cdcdcd",
                                  },
                                ]
                              }
                            >
                              Add Item
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                onLayout={[Function]}
                pointerEvents="box-none"
                style={
                  Object {
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "backgroundColor": "#FFF",
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    collapsable={undefined}
                    onLayout={[Function]}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "backgroundColor": "#FFF",
                        "borderBottomColor": "#A7A7AA",
                        "borderBottomWidth": 0.5,
                        "height": 64,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 20,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "bottom": 0,
                            "flexDirection": "row",
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      >
                        <View
                          collapsable={undefined}
                          pointerEvents="box-none"
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "bottom": 0,
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "left": 0,
                              "opacity": 1,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <Text
                            accessibilityTraits="header"
                            accessible={false}
                            allowFontScaling={true}
                            collapsable={undefined}
                            ellipsizeMode="tail"
                            numberOfLines={1}
                            onLayout={[Function]}
                            style={
                              Object {
                                "color": "rgba(0, 0, 0, .9)",
                                "fontSize": 17,
                                "fontWeight": "600",
                                "marginHorizontal": 16,
                                "textAlign": "center",
                              }
                            }
                          >
                            Add Item
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={undefined}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "#fff",
          "borderTopColor": "rgba(0, 0, 0, .3)",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 49,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel="Add Item"
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "height": 29,
              },
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#2f95dc",
                    "fontSize": 26,
                  },
                  undefined,
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#ccc",
                    "fontSize": 26,
                  },
                  undefined,
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          collapsable={undefined}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#3478f6",
              "fontSize": 11,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Add Item
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel="View Items"
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "height": 29,
              },
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#2f95dc",
                    "fontSize": 26,
                  },
                  undefined,
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#ccc",
                    "fontSize": 26,
                  },
                  undefined,
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          collapsable={undefined}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#929292",
              "fontSize": 11,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          View Items
        </Text>
      </View>
    </View>
  </View>
</View>
`;
